lib/streaming-pipeline.js

```diff
@@ -0,0 +1,193 @@
+const {Writable} = require('stream');
+const validateArgumentTransformers = require('./argument-transformer-validator');
+const RiffError = require('./riff-error');
+const OutputMarshaller = require('./output-marshaller');
+const logger = require('util').debuglog('riff');
+const InputUnmarshaller = require('./input-unmarshaller');
+const {guardWithTimeout} = require('./async');
+
+const toObject = (parameters) => {
+    const result = {};
+    for (let i = 0; i < parameters.length; i++) {
+        result[`${i}`] = parameters[i];
+    }
+    return result;
+};
+
+const validateArity = (arity) => {
+    if (typeof arity === 'undefined') {
+        throw new RiffError('error-function-arity', 'Cannot determine function arity. Aborting now');
+    }
+    if (typeof arity !== 'number' || Math.floor(arity) !== arity || arity <= 0) {
+        throw new RiffError('error-function-arity', `Function arity must be an integer >= 1, received: ${arity}. Aborting now`);
+    }
+};
+
+const validateTransformers = (transformers) => {
```

Comment:
why not call `validateArgumentTransformers` directly?

Reply:
`validateArgumentTransformers` assumption is that it must be called if and only if transformers are defined.